sale_order_import_invoice2data/wizard/sale_order_import_pdf.py

```diff
@@ -0,0 +1,178 @@
+            templates)
+        testspath = os.path.dirname(os.path.realpath(__file__))
+        templ_path = os.path.join(testspath, '../templates')
+        file_path = os.path.join(testspath, 'files', file_name)
+        templates += read_templates(templ_path)
+        try:
+            invoice2data_res = extract_data(file_name, templates=templates)
+        except Exception, e:
+            raise UserError(_(
+                "PDF saleorder parsing failed. Error message: %s") % e)
+        if not invoice2data_res:
+            raise UserError(_(
+                "This PDF saleorder doesn't match a known template of "
+                "the invoice2data lib."))
+        logger.info(
+            'Result of invoice2data PDF extraction: %s', invoice2data_res)
+        return self.invoice2data_to_parsed_order(invoice2data_res)
+
+    @api.model
+    def invoice2data_to_parsed_order(self, invoice2data_res):
+        precision = self.env['decimal.precision'].precision_get('Product UoS')
+        parsed_order = {
+            'partner': {
+                'vat': invoice2data_res.get('vat'),
+                'name': invoice2data_res.get('partner_name'),
+                'email': invoice2data_res.get('partner_email'),
+                'website': invoice2data_res.get('partner_website'),
+                'siren': invoice2data_res.get('siren'),
+            },
+            'currency': {
+                'iso': invoice2data_res.get('currency'),
+            },
+            'amount_total': invoice2data_res.get('amount'),
+            'saleorder_number': invoice2data_res.get('invoice_number'),
+            'date': invoice2data_res.get('date'),
+            'date_due': invoice2data_res.get('date_due'),
+            'date_start': invoice2data_res.get('date_start'),
+            'date_end': invoice2data_res.get('date_end'),
+            'description': invoice2data_res.get('description'),
+        }
+        if 'amount_untaxed' in invoice2data_res:
+            parsed_order['amount_untaxed'] = invoice2data_res['amount_untaxed']
+        if 'lines' in invoice2data_res:
+            parsed_order['lines'] = []
+            for i, line in enumerate(invoice2data_res['lines']):
+                vals = {
+                    # TODO: bdio.compare_lines() expects
+                    # this way,
+                    'product': {
+                        'desc': line.get('desc'),
+                        'code': line.get('code'),
+                        'ean13': line.get('ean13'),
+                    },
+                    # sale_order_import expects this way,
+                    'desc': line.get('desc'),
+                    'code': line.get('code'),
+                    'ean13': line.get('ean13'),
+                    # --------------
+                    'taxes': line.get('taxes'),
+                    'unit_price': line.get('unit_price'),
+                    'price': line.get('price'),
+                }
```

Comment:
I think it would be nice to add the intersection of `line.keys()` and `self.env['sale.order.line']._fields.keys()` to the line values too. This way, you simply have to use odoo names in your templates, and you get whatever exotic field you might want to use without having to write custom code